src/features/island/buildings/components/building/tent/TentModal.tsx

```diff
@@ -0,0 +1,126 @@
+import React, { useContext, useState } from "react";
+import { CloseButtonPanel } from "features/game/components/CloseablePanel";
+import { SUNNYSIDE } from "assets/sunnyside";
+import { SplitScreenView } from "components/ui/SplitScreenView";
+import { Context } from "features/game/GameProvider";
+import { useActor } from "@xstate/react";
+import { BumpkinParts, interpretTokenUri } from "lib/utils/tokenUriBuilder";
+import { BumpkinBox } from "./BumpkinBox";
+import { getBumpkinUrl } from "./lib/getBumpkinUrl";
+import { CONFIG } from "lib/config";
+
+interface Props {
+  defaultSelectedIndex?: number;
+  onClose: () => void;
+}
+
+const DEFAULT_BUMPKIN_ALLOWANCE = 1;
+
+const baseUrl =
+  CONFIG.NETWORK === "mainnet"
+    ? `https://bumpkins.io/#/bumpkins`
+    : `https://testnet.bumpkins.io/#/bumpkins`;
+
+export const TentModal: React.FC<Props> = ({
+  defaultSelectedIndex,
+  onClose,
+}) => {
+  const { gameService } = useContext(Context);
+  const [
+    {
+      context: { state, bumpkins },
+    },
+  ] = useActor(gameService);
+
+  const [selectedBumpkin, setSelectedBumpkin] = useState<{
+    equipped: BumpkinParts;
+    tokenId: number;
+  }>(interpretTokenUri(bumpkins[defaultSelectedIndex ?? 0].tokenURI));
+
+  if (!bumpkins) return null;
+
+  const placedTents = (state.buildings.Tent || []).length;
+  const allowedBumpkins = placedTents + DEFAULT_BUMPKIN_ALLOWANCE;
+
+  const farmingBumpkins = bumpkins.slice(0, allowedBumpkins);
+  const nonFarmingBumpkins = bumpkins.slice(allowedBumpkins);
+
+  const MainContent = () => (
+    <div className="flex flex-col space-y-4">
+      <div>
+        <p className="text-sm mb-1">Showing on Farm</p>
+        <div className="flex">
```

Comment:
This one isn't strictly necessary for now but no harm in putting it in.

```suggestion
        <div className="flex flex-wrap">
```